src/sys/resource.rs

```diff
@@ -218,21 +218,22 @@ pub fn getrlimit(resource: Resource) -> Result<(Option<rlim_t>, Option<rlim_t>)>
 ///
 /// * `resource`: The [`Resource`] that we want to set the limits of.
 /// * `soft_limit`: The value that the kernel enforces for the corresponding
-///   resource. Note: `None` input will be replaced by constant `RLIM_INFINITY`.
+///   resource.
 /// * `hard_limit`: The ceiling for the soft limit. Must be lower or equal to
-///   the current hard limit for non-root users. Note: `None` input will be
-///   replaced by constant `RLIM_INFINITY`.
+///   the current hard limit for non-root users.
 ///
-/// > Note: for some os (linux_gnu), setting hard_limit to `RLIM_INFINITY` can
-/// > results `EPERM` Error. So you will need to set the number explicitly.
+/// The special value `RLIM_INFINITY` indicates that no limit will be
+/// enforced. For some OSes (linux_gnu), setting the hard limit to
```

Comment:
I don't think that Linux returns EPERM because the user tried to set RLIM_INFINITY specifically.  Rather, I think that happens because the user tries to raise the limit.  That's disallowed on every OS.

Reply:
Yes, I wasn't sure why the comment was initially worded the way it was. I can fix it.

Reply:
Actually, is there a point in fixing it? It would just be duplicating the spec, that's already linked as a reference. Should I just delete that sentence?

Reply:
@asomers Ping.

Reply:
Removing it should be ok, but we shouldn't leave the wrong explanation in place.

Reply:
Done.